src/Access/UsersConfigAccessStorage.cpp

```diff
@@ -122,7 +122,9 @@ namespace
     {
         const bool validate = true;
         auto user = std::make_shared<User>();
-        user->setName(user_name);
+        String unescaped_user_name = user_name;
+        Poco::replaceInPlace(unescaped_user_name, "\\.", ".");
```

Comment:
Let's simply do replacement in the `user_name` with a proper comment

Reply:
Done. In order to do so, I had to change the function signature to take the `user_name` by value, otherwise `replaceInPlace` wouldn't work